File: code/Book.java
import java.util.Arrays;

// Total for Class: 25 Marks
public class Book {
	private String title;
	private String author;
	private String category;
	private String isbn;
	private double price;
	private int[] sales = new int[12];
	
	// 5 Marks
	public Book2(String title, String author, String category, String isbn, double price) {
		// TO-DO
	}
	
	public String getCategory() {
		return category;
	}
	
	public String getISBN() {
		return isbn;
	}
	
	// 3 Marks
	public void setSales(int month, int count) {
		// TO-DO
	}
	
	// 3 Marks
	public double getRevenue(int month) {
		// TO-DO
	}
	
	// 7 Marks
	public double getYearRevenue() {
		// TO-DO
	}
	
	// 7 Marks
	public String toString() {
		// TO-DO
	}
}

File: book.py

class Book:
    def __init__(self, isbn, title, author, category, price):
        self.title = title
        self.author = author
        self.category = category
        self.isbn = isbn
        self.price = float(price)
        self.sales = [0] * 12


    def get_category(self):
        return self.category

    def get_isbn(self):
        return self.isbn

    def set_sales(self, month, count):
        self.sales[month] = count

    def get_revenue(self, month):
        return self.price * self.sales[month]

    def get_year_revenue(self):
        return self.price * sum(self.sales)

    def to_string(self):
        return f"ISBN: {self.isbn} \nTitle and Author: {self.title}, {self.author} \nCategory: {self.category} \nPrice: R{self.price} \nSales: {self.sales}"



if __name__ == "__main__":
    pass
